services/vm-watcher/vm_watcher_service_test.go

```diff
@@ -356,6 +356,64 @@ var _ = Describe(
 					})))
 				})
 
+				When("a Zone that is being deleted", func() {
+					BeforeEach(func() {
+						outerInitEnvFn := initEnvFn
+
+						initEnvFn = func(ctx *builder.IntegrationTestContextForVCSim) {
+							outerInitEnvFn(ctx)
+
+							zone := &topologyv1.Zone{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "bogus",
+									Namespace: vcSimCtx.NSInfo.Namespace,
+									Finalizers: []string{
+										"fake/finalizer", // Finalizer to prevent deletion
+									},
+								},
+								Spec: topologyv1.ZoneSpec{},
+							}
+							Expect(ctx.Client.Create(ctx, zone)).To(Succeed())
+							// Delete the Zone so the deletion timestamp is set.
+							Expect(ctx.Client.Delete(ctx, zone)).To(Succeed())
+						}
+					})
+
+					When("the zone has the zone finalizer", func() {
```

Comment:
By `zone finalizer` do you mean the specific `zonectrl.Finalizer`? If so, I don't see it's being added anywhere in this block?

Reply:
Fixed this.  Can you PTAL?